### PR TITLE
Feature: Add collection

### PR DESCRIPTION
**Pull request by fernandes**
Imagine every time you want perform a search or a scope, you need to code on `process` method... not anymore!

This PR add support for scopes (thanks Ransack) and pagination (thanks Kaminari)

How does it work? On your controller to get a collection you do (for more info check https://github.com/apotonick/trailblazer/pull/36):

```ruby
def index
  collection Thing::Index
end
```

On your operation you need to:

```ruby
class Thing < ActiveRecord::Base
  class Index < Trailblazer::Operation
    # including CRUD and setting model is needed for both Scope and Pagination
    include CRUD, Scope, Pagination
    model Thing
  end
end
```

In your request you need to pass:

* `params[:q]` based on Ransack, check its [basic searching page](https://github.com/activerecord-hackery/ransack/wiki/Basic-Searching) for more information about it

* `params[:page]` for what page you want and `params[:per_page]` for how many results per page will be shown

And thats it... scope and pagination out of the box! :laughing:

### CODE REVIEW

**Comment by fernandes** [comment]
@apotonick This PR supress https://github.com/apotonick/trailblazer/pull/36, just left open to make easy to review the collection feature, merge, then review scope and search feature

they are split in two because collection was PRed before and is a pre requisite for scopes and search

**Comment by fernandes** [comment]
just keeping https://github.com/apotonick/trailblazer/pull/40

**Comment by halan** [comment]
:heart_eyes:

**Comment by apotonick** [comment]
@fernandes I think he loves you. :stuck_out_tongue_winking_eye:

**Comment by fernandes** [comment]
wtf brow... kkkkkkkkkkkkkkk

maybe @akitaonrails or @macedo can explain this :stuck_out_tongue_winking_eye:

**Comment by halan** [comment]
I need this feature :heart:, but about @fernandes, we can talk :heart:. @macedo dont care about.

**Comment by fernandes** [comment]
kkkkkkkkkkkkkk :love_letter:

**Comment by fernandes** [comment]
but about the feature... we are discussing how to set a more generic API, in a non-intrusive (don't explicit require kaminari and ransack)

probably soon its gonna be merged

**Comment by macedo** [comment]
@halan, don't embarrass me in front of my old friend @fernandes, please! kkkk

File: lib/trailblazer/operation/crud.rb [hunk 1]
```diff
@@ -62,6 +62,13 @@ def update_model(params)
 
       alias_method :find_model, :update_model
 
+      def all_model(params)
+        self.class.model_class.all
+      end
+      
+      def fetch(params)
```
Comment (fernandes) [inline lib/trailblazer/operation/crud.rb]: TODO: remove this

File: lib/trailblazer/operation/worker.rb [hunk 2]
```diff
@@ -74,7 +74,7 @@ def file_marshaller_representer
 
             # TODO: where do we set /tmp/uploads?
             dfn.merge!(
-              :serialize   => lambda { |file, *| Trailblazer::Operation::UploadedFile.new(file, :tmp_dir => "/tmp/uploads").to_hash },
+              :serialize   => lambda { |file, *| Trailblazer::Operation::UploadedFile.new(file, :tmp_dir => File.join(File.dirname(__FILE__), '../../..', 'tmp', 'uploads')).to_hash },
```
Comment (apotonick) [inline lib/trailblazer/operation/worker.rb]: Please remove this, this is wrong.

File: test/rails/controller_test.rb [hunk 3]
```diff
@@ -33,7 +33,7 @@ class ProcessParamsTest < ActionController::TestCase
   end
 
   test "Create with overridden #process_params" do
-    post :create, band: {name: "Kreator"}
+    post :create, band: {name: "Kreator", locality: "Essen"}
```
Comment (apotonick) [inline test/rails/controller_test.rb]: Please revert, this is the whole point of the test to **not** provide `locality`.

File: test/rails/controller_test.rb [hunk 4]
```diff
@@ -170,7 +170,7 @@ class ControllerPresentTest < ActionController::TestCase
 
     get :show, id: band.id
 
-    assert_equal "bands/show.html: Band,Band,true,Band::Update,Essen\n", response.body
+    assert_equal "bands/show.html: Band,Band,true,Band::Update,Nofx", response.body
```
Comment (apotonick) [inline test/rails/controller_test.rb]: Please revert, this is wrong.

**Comment by apotonick** [comment]
Where's the squashed and updated commit we discussed, sweetheart?

**Comment by fernandes** [comment]
@apotonick check #61 please

**Pull request closed without merging**